src/Serilog/Context/LogContext.cs

```diff
@@ -233,6 +233,11 @@ static ImmutableStack<ILogEventEnricher> Enrichers
             }
             set
             {
+                if (value == null)
+                {
+                    throw new ArgumentException();
```

Comment:
Should this check be here? (If the value is nullable, why not accept null?)

If the flow is correct - this should be `ArgumentNullException` I think.